### ISSUE TITLE
different weight limits for different truck types

### ISSUE THREAD

**Issue by johndh** [osmandapp/OsmAnd#13121]
# 🚀 feature request

Bridges in the USA often have different weight limits based on whether the vehicle is a straight truck, combination vehicle (tractor-trailer), or combination with multiple trailers, so there should be a way to specify what kind of truck I'm driving in my truck navigation profile. For example, a bridge sign might specify 13T for a straight truck and 23T for combination vehicles, and my truck is rated at 22T.

### Describe the solution you'd like
Among the other settings in the truck parameters, a selection for straight truck, combination, or multiple trailers. There may be better technical terms.

### Describe alternatives you've considered

Currently I have to know from experience which routes to avoid.

**Comment by vshcherb** [comment]
As workaround:  you can create different profiles and specify different Max weight for each of them. So in navigation you can select needed profile

**Comment by johndh** [comment]
> As workaround: you can create different profiles and specify different Max weight for each of them. So in navigation you can select needed profile

I don't understand how that is a workaround. How does OsmAnd know which weight limit of a bridge applies to the truck I'm driving if it doesn't know whether I'm driving a straight truck, combination, etc?

**Comment by vshcherb** [comment]
You change "Profile" manually - Truck / Car or custom profile. In each profile you have separate settings

![telegram-cloud-photo-size-4-5787541862872103146-y](https://user-images.githubusercontent.com/1042025/140981014-87f42311-bdd4-4982-b6c1-521db93776f8.jpg)

**Comment by jfburke3** [comment]
I don't think this can be addressed until the OSM community settles on exactly how to name these truck configurations.  They appear to have different name throughout the U.S.  There have been some attempts to start conversations on the topic, but they always stall out over what terms to use for the truck configurations.  It doesn't help that weight limits *also* depend on the axle configuration, not just the truck configuration (e.g., a truck with one axle at the front, one axle at the back of the cab area, and one axle at the back of the cargo compartment; vs. a truck with one axle at the front, and two axles at the back of the cargo compartment).  I've seen weight limit signs with 5 different truck/axle configurations on them.  One such sign:  https://www.mapillary.com/app/?pKey=465736701201924

**Comment by johndh** [comment]
> I don't think this can be addressed until the OSM community settles on exactly how to name these truck configurations. They appear to have different name throughout the U.S.

Without rehashing old debates about terminology, I think having a "trailer? yes/no" option would address most cases. Instead, having a "number of trailers? 0 to 3" option would address a few more. In my experience, I think that would be equivalent to other truck GPS apps/devices.
